### PR TITLE
Resources are now fetched from appropriate layout

### LINKED ISSUE #97: Resource editor should prefill all resources from layout

### PR DESCRIPTION
**Pull request by lauriLukkarinen**
- removed ability to create new resources to page
- refactored resources to be identified by their index in page resources list

Resolves #97

### CODE REVIEW

File: src/components/exhibition/exhibition-view.tsx [hunk 1]
```diff
@@ -189,16 +190,13 @@ export class ExhibitionView extends React.Component<Props, State> {
...
 
-          <ElementSettingsPane minWidth={ 320 } title={ selectedResource ? selectedResource.id : "" }>
+          <ElementSettingsPane minWidth={ 320 } title="">
             {
-              this.state.selectedResource &&
-              <ResourceEditor 
-                resource={ this.state.selectedResource }
-                selectedElement={ this.state.selectedElement }
-                onChange={ this.onResourceDataChange} />
+              selectedResourceIndex !== undefined &&
```
Comment (jarinykanen) [inline src/components/exhibition/exhibition-view.tsx:195]:
```suggestion
              selectedResourceIndex &&
```

File: src/components/exhibition/exhibition-view.tsx [hunk 2]
```diff
@@ -189,16 +190,13 @@ export class ExhibitionView extends React.Component<Props, State> {
...
+              this.renderResourceEditor()
             }
             {
-              this.state.selectedEventTriggerIndex !== undefined &&
+              selectedEventTriggerIndex !== undefined &&
```
Comment (jarinykanen) [inline src/components/exhibition/exhibition-view.tsx:199]:
```suggestion
              selectedEventTriggerIndex &&
```

File: src/components/exhibition/exhibition-view.tsx [hunk 3]
```diff
@@ -410,6 +408,35 @@ export class ExhibitionView extends React.Component<Props, State> {
     );
   }
 
+  /**
+   * Renders resource editor
+   */
+  private renderResourceEditor = () => {
+    const { selectedResourceIndex } = this.state;
+    if (selectedResourceIndex === undefined) {
```
Comment (jarinykanen) [inline src/components/exhibition/exhibition-view.tsx:416]:
```suggestion
    if (!selectedResourceIndex) {
```

File: src/utils/resource-utils.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import { ExhibitionPageResource, PageLayoutView, PageLayoutViewProperty, ExhibitionPageResourceType } from "../generated/client";
+
+export default class ResourceUtils {
```
Comment (jarinykanen) [inline src/utils/resource-utils.ts:3]: Add docs

File: src/utils/resource-utils.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { ExhibitionPageResource, PageLayoutView, PageLayoutViewProperty, ExhibitionPageResourceType } from "../generated/client";
+
```
Comment (lauriLukkarinen) [inline src/utils/resource-utils.ts:2]:
```suggestion
/**
 * Utility class for resources
 */
```

File: src/utils/resource-utils.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import { ExhibitionPageResource, PageLayoutView, PageLayoutViewProperty, ExhibitionPageResourceType } from "../generated/client";
+/**
```
Comment (lauriLukkarinen) [inline src/utils/resource-utils.ts:2]:
```suggestion

/**
```

**Pull request merged**